blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DBasicMetricsRW.java

```diff
@@ -0,0 +1,350 @@
+
+            LOG.error( msg );
+
+            throw new IOException( msg );
+        }
+
+        return super.getDataToRollup( locator, rollupType, range, columnFamilyName );
+    }
+
+    /**
+     * Returns true if the metric should be persisted.
+     *
+     * Blueflood can be configured not to save strings, and to only keep strings for specific tenants.  Fun!
+     *
+     * @param metric
+     * @return
+     */
+    private boolean shouldPersist( IMetric metric ) {
+
+        String tenantId = metric.getLocator().getTenantId();
+        boolean stringOrBool = DataType.isStringOrBoolean( metric.getMetricValue() );
+
+        if( stringOrBool && areStringMetricsDropped && !keptTenantIdsSet.contains( tenantId ) ) {
+            return false;
+        }
+        else if( stringOrBool ) {
+
+            String currentValue = String.valueOf(metric.getMetricValue());
+            final String lastValue = rawIO.getLastStringValue( metric.getLocator() );
+
+            return lastValue == null || !currentValue.equals(lastValue);
+        }
+
+        return true;
+    }
+
+    /**
+     *  Return map of MetricDatas for string values in the given range.
+     *
+     * @param locators
+     * @param range
+     * @return
+     */
+    private Map<Locator, MetricData> getStringDataForRange( List<Locator> locators, Range range ) {
+
+        return getStringBooleanDataForRange( locators, range, false );
+    }
+
+    /**
+     * Return map of MetricDatas for strings or booleans in the given range, depending on the isBoolean param.
+     *
+     * @param locators
+     * @param range
+     * @param isBoolean if true, return MetricDatas as type Boolean
+     * @return
+     */
+    private Map<Locator, MetricData> getStringBooleanDataForRange( List<Locator> locators, Range range, boolean isBoolean ) {
+
+        Map<Locator, MetricData> metrics = new HashMap<Locator, MetricData>();
+        Map<Locator, ResultSetFuture> futures = new HashMap<Locator, ResultSetFuture>();
+
+        for( Locator locator : locators ) {
+            futures.put( locator, rawIO.getStringAsync( locator, range ) );
+        }
+
+        for( Map.Entry<Locator, ResultSetFuture> future : futures.entrySet() ) {
+
+            Points points = new Points<String>();
+
+            try {
+                for ( Row row : future.getValue().getUninterruptibly().all() ) {
+
+                    Object value = isBoolean ? Boolean.valueOf( row.getString( VALUE ) ) : row.getString( VALUE );
+                    points.add( new Points.Point( row.getLong( COLUMN1 ), value ) );
+                }
+
+                MetricData.Type type = isBoolean ? MetricData.Type.BOOLEAN : MetricData.Type.BOOLEAN;
+
+                metrics.put( future.getKey(), new MetricData( points, getUnitString( future.getKey() ), type ) );
+            }
+            catch (Exception e ) {
+
+                LOG.error( String.format( "error writing metric %s, metrics_string", future.getKey() ), e );
```

Comment:
Blueflood doesn't throw exceptions on reads, it logs the error and silently continues.  I don't think we should change this now.